determined-ai/determined
# fix: cursor in modal text field jumps to end of input

## Description
[DET-7392] [DET-7225]




## Test Plan
- Check if cursor won't jump to the end of text in `new Model modal`, `new project modal` in workspace and `new Workspace modal`
- Check if warning will emerge when there is a duplicate name in models
- Check if we can create model in model registry with name substring of existing model




## Commentary (optional)
- newly added `react-hooks-form` to fix this issue and for future form use. This library is better than other libraries like formik, react form and more in terms of performance and bundle size. What do you think?



## Checklist

See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[DET-7392]: https://determinedai.atlassian.net/browse/DET-7392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DET-7225]: https://determinedai.atlassian.net/browse/DET-7225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
